src/CDataInterface/format_string_to_types.jl

```diff
@@ -0,0 +1,68 @@
+# https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings
+module FormatStrings
```

Comment:
This might be more of a personal style, but I wouldn't wrap this in a module here. In my mind, submodules are like mini packages that might be useful to split out on their own one day with stand-alone functionality. In this case, the format parsing stuff is pretty closely tied to the arrowschema, so I think it's better to just `include` the file.